integrations/kredis/set.go

```diff
@@ -0,0 +1,66 @@
+package kredis
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/keploy/go-sdk/keploy"
+	"go.keploy.io/server/pkg/models"
+)
+
+// For Set method
+
+type KStatusCmd struct {
```

Comment:
any reason why we didn't use the `Cmdable` interface as mentioned here? - https://github.com/go-redis/redis/issues/515

Reply:
<img width="686" alt="Screenshot 2022-09-16 at 12 22 46 PM" src="https://user-images.githubusercontent.com/42648568/190579829-bf940368-ca15-411d-a2e6-351996b4dca0.png">

Here the Cmdable interface is being implemented by cmdable field of Client and it is assigned with Process method func of Client. Now, since we can't assign the reciever methods of a struct and there are no methods to update the cmdable field of redis.Client. Thus, embedded redis.Client in keploy struct.
Also the issue in the redis github, shows that if the user stores the redis.Client in redis.Cmdable typed variable. Also redis.Client have some methods other than redis.Cmdable interface.

Reply:
As discussed, to get around this we can build against the go-redis universal client.